Cannot install swc_cli on stable Rust

### Describe the bug

Hi there,

I'm new to Rust + interested in writing a SWC plugin.

Unfortunately, following the [instructions on the website](https://swc.rs/docs/plugin/ecmascript/getting-started) I got the following output when running `cargo install swc_cli`

```
╰─ cargo install swc_cli
    Updating crates.io index
  Downloaded swc_cli v0.91.71
  Downloaded 1 crate (38.8 KB) in 0.97s
  Installing swc_cli v0.91.71
    Updating crates.io index
(shortened for brevity...)
   Compiling walkdir v2.4.0
   Compiling swc_ecma_utils v0.123.0
   Compiling clap v3.2.25
   Compiling path-absolutize v3.1.1
   Compiling tracing-futures v0.2.5
   Compiling tracing-chrome v0.5.0
   Compiling relative-path v1.9.0
   Compiling swc_ecma_transforms_base v0.133.2
   Compiling swc_ecma_usage_analyzer v0.19.0
   Compiling swc_ecma_ext_transforms v0.109.0
   Compiling swc_ecma_lints v0.88.2
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> /Users/gwong2/.cargo/registry/src/index.crates.io-6f17d22bba15001f/swc_ecma_lints-0.88.2/src/lib.rs:3:1
  |
3 | #![feature(box_patterns)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0554`.
error: could not compile `swc_ecma_lints` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `swc_cli v0.91.71`, intermediate artifacts can be found at `/var/folders/tf/lrbmvpzj3wd7_kn0qrvl6yjc0000gn/T/cargo-install4ouVSC`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

Is there a way to install `swc_cli` without switching to nightly Rust?

### Input code

_No response_

### Config

_No response_

### Playground link

_No response_

### SWC Info output

_No response_

### Expected behavior

`swc_cli` can be installed with stable Rust without issue

### Actual behavior

`swc_cli` cannot be installed on stable Rust

### Version

swc_cli v0.91.71

```
╰─ rustc --version
rustc 1.72.0 (5680fa18f 2023-08-23)
```

### Additional context

_No response_

--- a/crates/swc_ecma_lints/src/lib.rs
+++ b/crates/swc_ecma_lints/src/lib.rs
@@ -1,6 +1,5 @@
 #![cfg_attr(feature = "non_critical_lints", deny(unused))]
 #![cfg_attr(feature = "non_critical_lints", deny(clippy::all))]
-#![feature(box_patterns)]
 
 pub mod config;
 pub mod rule;

--- a/crates/swc_ecma_lints/src/rules/no_alert.rs
+++ b/crates/swc_ecma_lints/src/rules/no_alert.rs
@@ -129,6 +129,37 @@ impl NoAlert {
         }
     }
 
+    fn handle_member_expr(&mut self, member_expr: &MemberExpr) {
+        let MemberExpr { obj, prop, .. } = member_expr;
+
+        match obj.as_ref() {
+            Expr::Ident(obj) => {
+                if !self.is_satisfying_indent(obj) {
+                    return;
+                }
+
+                self.obj = Some(obj.sym.clone());
+
+                self.handle_member_prop(prop);
+            }
+            Expr::This(_) => {
+                let inside_arrow_fn = self.is_inside_arrow_fn();
+                let inside_class = self.is_inside_class();
+
+                if inside_arrow_fn && inside_class {
+                    return;
+                }
+
+                if !inside_arrow_fn && (inside_class || self.is_inside_object()) {
+                    return;
+                }
+
+                self.handle_member_prop(prop);
+            }
+            _ => {}
+        }
+    }
+
     fn handle_callee(&mut self, expr: &Expr) {
         match expr {
             Expr::Ident(ident) => {
@@ -136,39 +167,10 @@ impl NoAlert {
                     self.prop = Some(ident.sym.clone());
                 }
             }
-            Expr::Member(member_expr)
-            | Expr::OptChain(OptChainExpr {
-                base: box OptChainBase::Member(member_expr),
-                ..
-            }) => {
-                let MemberExpr { obj, prop, .. } = member_expr;
-
-                match obj.as_ref() {
-                    Expr::Ident(obj) => {
-                        if !self.is_satisfying_indent(obj) {
-                            return;
-                        }
-
-                        self.obj = Some(obj.sym.clone());
-
-                        self.handle_member_prop(prop);
-                    }
-                    Expr::This(_) => {
-                        let inside_arrow_fn = self.is_inside_arrow_fn();
-                        let inside_class = self.is_inside_class();
-
-                        if inside_arrow_fn && inside_class {
-                            return;
-                        }
-
-                        if !inside_arrow_fn && (inside_class || self.is_inside_object()) {
-                            return;
-                        }
-
-                        self.handle_member_prop(prop);
-                    }
-                    _ => {}
-                }
+            Expr::Member(member_expr) => self.handle_member_expr(member_expr),
+            Expr::OptChain(OptChainExpr { base, .. }) if base.is_member() => {
+                let member_expr = base.as_member().unwrap();
+                self.handle_member_expr(member_expr);
             }
             Expr::OptChain(opt_chain) => {
                 opt_chain.visit_children_with(self);

--- a/crates/swc_ecma_lints/src/rules/radix.rs
+++ b/crates/swc_ecma_lints/src/rules/radix.rs
@@ -243,11 +243,10 @@ impl Radix {
             Expr::Member(member_expr) => {
                 return self.extract_obj_and_prop_member_case(member_expr);
             }
-            Expr::OptChain(OptChainExpr {
-                base: box OptChainBase::Member(member_expr),
-                ..
-            }) => {
-                return self.extract_obj_and_prop_member_case(member_expr);
+            Expr::OptChain(OptChainExpr { base, .. }) => {
+                if let OptChainBase::Member(member_expr) = &**base {
+                    return self.extract_obj_and_prop_member_case(member_expr);
+                }
             }
             Expr::Paren(ParenExpr { expr, .. }) => {
                 return self.extract_obj_and_prop(expr.as_ref());

